packages/vscode-extension/src/debug/progressHelper.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IProgressHandler } from "@microsoft/teamsfx-api";
+
+export class ParallelProgressHelper {
+  private details: string[];
+  constructor(private progressBar: IProgressHandler) {
+    this.details = [];
+  }
+
+  public async start(details: string[]): Promise<void> {
+    this.details = details;
+  }
+
+  public async next(): Promise<void> {
+    if (this.details.length == 0) {
+      return;
+    }
+    const detail = this.details.shift();
```

Comment:
How to ensure this `next()` is called in correct order?

E.g.,
- When starting, `details` are `["frontend","backend","bot"]`
- Then, bot installation finished, `details` becomes `["backend","bot"]`?

Reply:
Progress is for UI part so better to modify checker logic as less as possible. Considering such `ProgressHelper`:
``` typescript
class ProgressHelper {
  steps: [{name, message, finished}];
  currentStep = 0;
  construct(allSteps: []);
  done(stepName) {
    step[stepName].finished = true;
    while(steps[currentStep].finished) {
      currentStep++;
      progressBar.next(steps[currentStep].message);
    }
  }
}
```
Then only need to call `ProgressHelper.done(checkerName)` after each checker finishes.